zerver/lib/typed_endpoint.py

```diff
@@ -0,0 +1,514 @@
+                func_param.default is NotSpecified
+            ), f"Path-only parameter {func_param.param_name} should not have a default value"
+        # Record arguments that should be documented so that our
+        # automated OpenAPI docs tests can compare these against the code.
+        if (
+            func_param.documentation_status is DocumentationStatus.DOCUMENTED
+            and not func_param.path_only
+        ):
+            # TODO: Move arguments_map to here once zerver.lib.request does not
+            # need it anymore.
+            arguments_map[endpoint_info.view_func_full_name].append(func_param.request_var_name)
+
+    @wraps(view_func)
+    def _wrapped_view_func(
+        request: HttpRequest, /, *args: ParamT.args, **kwargs: ParamT.kwargs
+    ) -> ReturnT:
+        request_notes = RequestNotes.get_notes(request)
+        for parameter in endpoint_info.parameters:
+            if parameter.path_only:
+                # For path_only parameters, they should already have been passed via
+                # the URL, so there's no need for us to do anything.
+                #
+                # TODO: Run validators for path_only parameters for NewType.
+                assert (
+                    parameter.param_name in kwargs
+                ), f"Path-only variable {parameter.param_name} should be passed already"
+            if parameter.param_name in kwargs:
+                # Skip parameters that are already supplied by the caller.
+                continue
+
+            # Extract the value to parse from the request body if specified.
+            if parameter.argument_type_is_body:
+                try:
+                    request_notes.processed_parameters.add(parameter.request_var_name)
+                    kwargs[parameter.param_name] = parse_value_for_parameter(
+                        parameter, request.body.decode(request.encoding or "utf-8")
+                    )
+                except UnicodeDecodeError:
+                    raise JsonableError(_("Malformed payload"))
+                # test_typed_endpoint.TestEndpoint.test_argument_type has
+                # coverage of this, but coverage.py fails to recognize it for
+                # some reason.
+                continue  # nocoverage
+
+            # Otherwise, try to find the matching request variable in one of the QueryDicts
+            # This is a view bug, not a user error, and thus should throw a 500.
+            possible_aliases = [parameter.request_var_name, *parameter.aliases]
+            alias_used = None
+            value_to_parse = None
+
+            for current_alias in possible_aliases:
+                if current_alias in request.POST:
+                    value_to_parse = request.POST[current_alias]
+                elif current_alias in request.GET:
+                    value_to_parse = request.GET[current_alias]
+                else:
+                    # This is covered by test_REQ_aliases, but coverage.py
```

Comment:
The test name here looks copied from the original `REQ` implementation.

Reply:
Indeed, should be `test_aliases` from the other file.